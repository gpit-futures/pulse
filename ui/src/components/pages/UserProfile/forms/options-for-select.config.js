const optionsForThemesField = [{
  value: 'green',
  title: 'Green Theme',
},
{
  value: 'purple',
  title: 'Purple Theme',
},
{
  value: 'gray',
  title: 'Gray Theme',
}];

const optionsForGenderField = [{
  value: 'female',
  title: 'Female',
}, {
  value: 'male',
  title: 'Male',
}];

const optionsForCountryField = [{
  value: 'usa',
  title: 'USA',
}, {
  value: 'country1',
  title: 'Country 1',
}, {
  value: 'country2',
  title: 'Country 2',
}, {
  value: 'country3',
  title: 'Country 3',
}, {
  value: 'country4',
  title: 'Country 4',
}, {
  value: 'country5',
  title: 'Country 5',
}];

export { optionsForThemesField, optionsForGenderField, optionsForCountryField }
