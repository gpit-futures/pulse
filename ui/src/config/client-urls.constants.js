import { themeConfigs } from '../themes.config';

export const clientUrls = {
  ROOT: '/',
  CHARTS: '/charts',
  PATIENTS: '/patients',
  PATIENTS_FULL_DETAILS: '/patients-full-details',
  SEARCH_REPORT: '/search-report',
  USER_PROFILE: '/profile',
  PATIENTS_SUMMARY: 'patients-summary',
  RESULTS: 'results',
  DIAGNOSES: 'diagnoses',
  ALLERGIES: 'allergies',
  CONTACTS: 'contacts',
  CLINICAL_NOTES: 'clinicalNotes',
  PERSONAL_NOTES: 'personalNotes',
  VACCINATIONS: 'vaccinations',
  MEDICATIONS: 'medications',
  GENERIC_PLUGIN: 'genericPlugin',
  PROCEDURES: 'procedures',
  EVENTS: 'events',
  TEST_RESULTS: 'results',
  REFERRALS: 'referrals',
  ORDERS: 'orders',
  MDTS: 'mdt',
  DRAWINGS: 'drawings',
  VITALS: 'vitals',
  DOCUMENTS: 'documents',
  IMAGES: 'images',
  TRANSFERS_OF_CARE: 'transfer-of-care',
  TOP_THREE_THINGS: 'topThreeThings',
  CLINICAL_STATEMENTS: 'clinicalStatements',
  PROMS: 'proms',
  FEEDS: 'feeds',
};

const pluginsPages = {
  'allergies': {
    breadcrumbs: [{
      title: 'Allergies',
      state: '/allergies',
    }],
  },
  'diagnoses': {
    breadcrumbs: [{
      title: 'Problems / Diagnoses',
      state: '/diagnoses',
    }],
  },
  'clinicalNotes': {
    breadcrumbs: [{
      title: 'Clinical Notes',
      state: '/clinicalNotes',
    }],
  },
  'personalNotes': {
    breadcrumbs: [{
      title: 'Personal Notes',
      state: '/personalNotes',
    }],
  },
  'contacts': {
    breadcrumbs: [{
      title: 'Contacts',
      state: '/contacts',
    }],
  },
  'vaccinations': {
    breadcrumbs: [{
      title: 'Vaccinations',
      state: '/vaccinations',
    }],
  },
  'genericPlugin': {
    breadcrumbs: [{
      title: 'Generic Plugin',
      state: '/genericPlugin',
    }],
  },
  'medications': {
    breadcrumbs: [{
      title: 'Medications',
      state: '/medications',
    }],
  },
  'procedures': {
    breadcrumbs: [{
      title: 'Procedures',
      state: '/procedures',
    }],
  },
  'events': {
    breadcrumbs: [{
      title: 'Events',
      state: '/events',
    }],
  },
  'results': {
    breadcrumbs: [{
      title: 'Test Results',
      state: '/results',
    }],
  },
  'referrals': {
    breadcrumbs: [{
      title: 'Referrals',
      state: '/referrals',
    }],
  },
  'orders': {
    breadcrumbs: [{
      title: 'Orders',
      state: '/orders',
    }],
  },
  'mdt': {
    breadcrumbs: [{
      title: 'MDT',
      state: '/mdt',
    }],
  },
  'drawings': {
    breadcrumbs: [{
      title: 'Drawings',
      state: '/drawings',
    }],
  },
  'vitals': {
    breadcrumbs: [{
      title: 'Vitals - NEWS',
      state: '/vitals',
    }],
  },
  'documents': {
    breadcrumbs: [{
      title: 'Documents',
      state: '/documents',
    }],
  },
  'images': {
    breadcrumbs: [{
      title: 'Images',
      state: '/images',
    }],
  },
  'transfer-of-care': {
    breadcrumbs: [{
      title: 'Transfers Of Care',
      state: '/transfers-of-care',
    }],
  },
  'topThreeThings': {
    breadcrumbs: [{
      title: 'Top 3 Things',
      state: '/topThreeThings',
    }],
  },
  'clinicalStatements': {
    breadcrumbs: [{
      title: 'Clinical Statements',
      state: '/clinicalStatements',
    }],
  },
  'proms': {
    breadcrumbs: [{
      title: 'PROMs',
      state: '/proms',
    }],
  },
};


const addPluginsPagesToLists = (list, breadcrumbsBefore) => {
  for (const nameOfPage in pluginsPages) {
    const breadcrumbs = breadcrumbsBefore.concat(pluginsPages[nameOfPage].breadcrumbs);
    list[nameOfPage] = { breadcrumbs };
  }
};

export const mainPagesTitles = {
  '/': {
    breadcrumbs: [{
      title: 'System Dashboard',
      state: '/',
    }],
    headerTitle: 'System Dashboard',
  },
  'charts': {
    breadcrumbs: [{
      title: 'System Dashboard',
      state: '/charts',
    }],
    headerTitle: 'System Dashboard',
  },
  'patients': {
    breadcrumbs: [{
      title: 'Home',
      state: '/',
    }, {
      title: 'Patient Listings',
      state: '/patients',
    }],
    headerTitle: 'Patients Lists',
  },
  'profile': {
    breadcrumbs: [{
      title: 'Patient Information',
      state: '/profile',
    }],
    headerTitle: 'Personal Information',
  },
  'search-report': {
    breadcrumbs: [{
      title: 'Home',
      state: '/',
    }, {
      title: 'Search Report',
      state: '/search-report',
    }],
    headerTitle: 'System Dashboard',
  },
  'patients-summary': {
    breadcrumbs: [{
      title: 'Patient Listings',
      state: '/patients',
    }, {
      title: themeConfigs.patientsSummaryPageName,
      state: '/patients-summary',
    }],
  },
  'patients-full-details': {
    breadcrumbs: [],
    headerTitle: 'Search results',
  },
};
addPluginsPagesToLists(mainPagesTitles, [{
  title: 'Patient Listings',
  state: '/patients',
}, {
  title: themeConfigs.patientsSummaryPageName,
  state: '/patients-summary',
}]);

export const mainPagesTitlesForPatients = {
  '/': {
    breadcrumbs: [{
      title: 'Patient Summary',
      state: '/patients-summary',
    }],
    headerTitle: 'Patient Summary',
  },

  'profile': {
    breadcrumbs: [{
      title: 'Patient Information',
      state: '/profile',
    }],
    headerTitle: 'Personal Information',
  },
  'patients-summary': {
    breadcrumbs: [{
      title: themeConfigs.patientsSummaryPageName,
      state: '/patients-summary',
    }],
  },
};
addPluginsPagesToLists(mainPagesTitlesForPatients, [{
  title: themeConfigs.patientsSummaryPageName,
  state: '/patients-summary',
}]);
