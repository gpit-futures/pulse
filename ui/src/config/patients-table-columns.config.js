import React from 'react';

import { getDDMMMYYYY } from '../utils/time-helpers.utils';

const CalendarIcon = () => <i className="fa fa-calendar" />;

export const patientsColumnsConfig = [
  { key: 'name', title: 'Name', width: 150 },
  { key: 'address', title: 'Address', width: 300 },
  { key: 'dateOfBirth', title: 'Born', transformer: getDDMMMYYYY, width: 105 },
  { key: 'promsCount', title: 'PROM ', icon: <span>#</span>, width: 100 },
  { key: 'promsDate', title: 'PROM', icon: <CalendarIcon />, transformer: getDDMMMYYYY, width: 110 },
  { key: 'gender', title: 'Gender', width: 90 },
  { key: 'id', title: 'NHS No.', width: 115 },
  { key: 'ordersDate', title: 'Orders', icon: <CalendarIcon />, transformer: getDDMMMYYYY, width: 110 },
  { key: 'ordersCount', title: 'Orders ', icon: <span>#</span>, width: 100 },
  { key: 'resultsDate', title: 'Results', icon: <CalendarIcon />, transformer: getDDMMMYYYY, width: 110 },
  { key: 'resultsCount', title: 'Results ', icon: <span>#</span>, width: 100 },
  { key: 'vitalsDate', title: 'Vitals', icon: <CalendarIcon />, transformer: getDDMMMYYYY, width: 110 },
  { key: 'vitalsCount', title: 'Vitals ', icon: <span>#</span>, width: 100 },
  { key: 'diagnosesDate', title: 'Diagnosis', icon: <CalendarIcon />, transformer: getDDMMMYYYY, width: 130 },
  { key: 'diagnosesCount', title: 'Diagnosis ', icon: <span>#</span>, width: 120 },
  { key: 'viewPatientNavigation', title: '', icon: <span />, width: 90 },
];

export const defaultColumnsSelected = {
  name: true,
  gender: true,
  dateOfBirth: true,
  address: true,
  id: true,
  ordersDate: true,
  resultsDate: true,
  vitalsDate: false,
  diagnosesDate: false,
  ordersCount: true,
  resultsCount: true,
  vitalsCount: false,
  diagnosesCount: false,
  viewPatientNavigation: true,
  promsCount: false,
  promsDate: false,
};
