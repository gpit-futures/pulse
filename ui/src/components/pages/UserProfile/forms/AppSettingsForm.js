import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import FileInput from '../../../form-fields/FileInput';
import { optionsForThemesField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';
import { validateAppSettingsForm } from '../forms.validation';
import { valuesSettingsForm, valuesSettingsFormLabels } from '../forms.config';
import { setTheme } from '../../../../ducks/set-theme.duck';

@reduxForm({
  form: 'appSettingsFormSelector',
  validate: validateAppSettingsForm,
})
export default class AppSettingsForm extends PureComponent {
  static propTypes = {
    dispatch: PropTypes.func,
    initialize: PropTypes.func,
  };

  componentDidMount() {
    const { patientsInfo, initialize } = this.props;
    initialize(this.defaultValuesForm(patientsInfo));
  }

  defaultValuesForm(value) {
    const defaultFormValues = {
      [valuesSettingsForm.APP_TITLE]: value.title,
      [valuesSettingsForm.LOGO_PATH]: value.logoB64,
      [valuesSettingsForm.SELECT_THEME]: value.themeColor,
      [valuesSettingsForm.BROWSER_TITLE]: value.browserTitle,
    };

    return defaultFormValues;
  }

 setThemeHook = dispatch => (theme) => {
   dispatch(setTheme(theme));
   return theme;
 };

 render() {
   const { dispatch } = this.props;

   return (
     <div className="panel-body-inner">
       <form name="appSettingsForm" className="form">
         <div className="form-group-wrapper">
           <Row>
             <Col xs={12} md={6}>
               <Row>
                 <Col md={11}>
                   <Field
                     label={valuesSettingsFormLabels.APP_TITLE}
                     name={valuesSettingsForm.APP_TITLE}
                     type="text"
                     placeholder=""
                     component={ValidatedInput}
                   />
                   <Field
                     label={valuesSettingsFormLabels.LOGO_PATH}
                     name={valuesSettingsForm.LOGO_PATH}
                     component={FileInput}
                     id={valuesSettingsForm.LOGO_PATH}
                   />
                   <Field
                     label={valuesSettingsFormLabels.SELECT_THEME}
                     name={valuesSettingsForm.SELECT_THEME}
                     component={Select}
                     options={optionsForThemesField}
                     //here we're using normalize hook for dispatching action to set app theme
                     normalize={this.setThemeHook(dispatch)}
                   />
                   <Field
                     label={valuesSettingsFormLabels.BROWSER_TITLE}
                     name={valuesSettingsForm.BROWSER_TITLE}
                     type="text"
                     placeholder=""
                     component={ValidatedInput}
                   />
                 </Col>
               </Row>
             </Col>
           </Row>
         </div>
       </form>
     </div>)
 }
}
