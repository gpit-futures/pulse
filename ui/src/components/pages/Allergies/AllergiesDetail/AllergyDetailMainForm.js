import React, { PureComponent } from 'react';
import { Field, reduxForm } from 'redux-form'

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import ValidatedTextareaFormGroup from '../../../form-fields/ValidatedTextareaFormGroup';
import DateInput from '../../../form-fields/DateInput';
import { validateAllergiesPanel } from '../forms.validation';
import { valuesNames, valuesLabels } from '../forms.config';

@reduxForm({
  form: 'allergiePanelFormSelector',
  validate: validateAllergiesPanel,
})
export default class AllergyDetailMainForm extends PureComponent {
  componentDidMount() {
    const { detail, initialize } = this.props;
    initialize(this.defaultValuesForm(detail));
  }
  defaultValuesForm(value) {
    const defaultFormValues = {
      [valuesNames.CAUSE]: value[valuesNames.CAUSE],
      [valuesNames.REACTION]: value[valuesNames.REACTION],
      [valuesNames.AUTHOR]: value[valuesNames.AUTHOR],
    };

    return defaultFormValues;
  }
  render() {
    const { detail, isSubmit } = this.props;
    return (
      <div className="panel-body-inner">
        <form name="allergiePanelForm" className="form">
          <div className="form-group-wrapper">
            <div className="row-expand">
              <div className="col-expand-left">
                <Field
                  label={valuesLabels.CAUSE}
                  name={valuesNames.CAUSE}
                  id={valuesNames.CAUSE}
                  type="text"
                  placeholder=""
                  component={ValidatedInput}
                  props={{ isSubmit }}
                />
                <Field
                  label={valuesLabels.REACTION}
                  name={valuesNames.REACTION}
                  id={valuesNames.REACTION}
                  component={ValidatedTextareaFormGroup}
                  props={{ isSubmit }}
                />
              </div>
            </div>

            {detail[valuesNames.ISIMPORT] ?
              <Field
                label={valuesLabels.IMPORT}
                name={valuesNames.IMPORT}
                id={valuesNames.IMPORT}
                component={ValidatedInput}
                props={{ disabled: true, isSubmit }}
              />
              : null
            }

            <div className="row-expand">
              <div className="col-expand-left">
                <Field
                  label={valuesLabels.AUTHOR}
                  name={valuesNames.AUTHOR}
                  id={valuesNames.AUTHOR}
                  component={ValidatedInput}
                  props={{ disabled: true }}
                />
              </div>
              <div className="col-expand-right">
                <Field
                  label={valuesLabels.DATE}
                  name={valuesNames.DATE}
                  id={valuesNames.DATE}
                  component={DateInput}
                  props={{ disabled: true, value: detail[valuesNames.DATE_CREATED], format: 'DD-MMM-YYYY' }}
                />
              </div>
            </div>
          </div>
        </form>
      </div>)
  }
}
