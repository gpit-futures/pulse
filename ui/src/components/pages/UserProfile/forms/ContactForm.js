import React, { PureComponent } from 'react';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import { optionsForCountryField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';
import { validateContactForm } from '../forms.validation';
import { valuesContactForm, valuesContactFormLabels } from '../forms.config';
import { defaultContactFormValues } from './default-values.config';

@reduxForm({
  form: 'contactFormSelector',
  validate: validateContactForm,
})
export default class ContactForm extends PureComponent {
  componentDidMount() {
    this.props.initialize(defaultContactFormValues);
  }

  render() {
    return (
      <div className="panel-body-inner">
        <form name="contactForm" className="form">
          <div className="form-group-wrapper">
            <Row>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11}>
                    <Field
                      label={valuesContactFormLabels.ADDRESS}
                      name={valuesContactForm.ADDRESS}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.CITY}
                      name={valuesContactForm.CITY}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.STATE}
                      name={valuesContactForm.STATE}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.POSTAL_CODE}
                      name={valuesContactForm.POSTAL_CODE}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.SELECT_COUNTRY}
                      name={valuesContactForm.SELECT_COUNTRY}
                      placeholder=""
                      component={Select}
                      options={optionsForCountryField}
                    />
                  </Col>
                </Row>
              </Col>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11} mdOffset={1}>
                    <Field
                      label={valuesContactFormLabels.PHONE}
                      name={valuesContactForm.PHONE}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.EMAIL}
                      name={valuesContactForm.EMAIL}
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
            </Row>
          </div>
        </form>
      </div>)
  }
}
