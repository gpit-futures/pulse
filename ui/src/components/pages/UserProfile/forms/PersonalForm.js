import React, { PureComponent } from 'react';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import DateInput from '../../../form-fields/DateInput';
import { optionsForGenderField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';
import { validatePersonalForm } from '../forms.validation';
import { valuesPersonalForm, valuesPersonalFormLabels } from '../forms.config';
import { defaultPersonalFormValues } from './default-values.config';

@reduxForm({
  form: 'personalFormSelector',
  validate: validatePersonalForm,
})
export default class PersonalForm extends PureComponent {
  componentDidMount() {
    this.props.initialize(defaultPersonalFormValues);
  }

  render() {
    return (
      <div className="panel-body-inner">
        <form name="personalForm" className="form">
          <div className="form-group-wrapper">
            <Row>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11}>
                    <Field
                      label={valuesPersonalFormLabels.FIRST_NAME}
                      name={valuesPersonalForm.FIRST_NAME}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesPersonalFormLabels.LAST_NAME}
                      name={valuesPersonalForm.LAST_NAME}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesPersonalFormLabels.NHS_NUMBER}
                      name={valuesPersonalForm.NHS_NUMBER}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11} mdOffset={1}>
                    <Field
                      label={valuesPersonalFormLabels.DATE_OF_BIRTH}
                      name={valuesPersonalForm.DATE_OF_BIRTH}
                      type="text"
                      placeholder=""
                      component={DateInput}
                      props={{ format: 'DD-MMM-YYYY' }}
                    />
                    <Field
                      label={valuesPersonalFormLabels.SELECT_GENDER}
                      name={valuesPersonalForm.SELECT_GENDER}
                      placeholder=""
                      component={Select}
                      options={optionsForGenderField}
                    />
                    <Field
                      label={valuesPersonalFormLabels.DOCTOR}
                      name={valuesPersonalForm.DOCTOR}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
            </Row>
          </div>
        </form>
      </div>)
  }
}
