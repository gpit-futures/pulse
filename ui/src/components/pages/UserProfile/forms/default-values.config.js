import { valuesSettingsForm, valuesPersonalForm, valuesContactForm } from '../forms.config';

const defaultPersonalFormValues = {
  [valuesPersonalForm.FIRST_NAME]: 'Bob',
  [valuesPersonalForm.LAST_NAME]: 'Smith',
  [valuesPersonalForm.SELECT_GENDER]: 'female',
  [valuesPersonalForm.DOCTOR]: 'Dr Emma Huston',
  [valuesPersonalForm.DATE_OF_BIRTH]: 0,
};

const defaultContactFormValues = {
  [valuesContactForm.ADDRESS]: '6801 Tellus Street',
  [valuesContactForm.CITY]: 'Westmorland',
  [valuesContactForm.STATE]: 'Westmorland',
  [valuesContactForm.POSTAL_CODE]: 'Box 306',
  [valuesContactForm.PHONE]: '07624 647524',
  [valuesContactForm.EMAIL]: 'bob.smith@gmail.com',
};

const defaultAppSettingsFormValues = {
  [valuesSettingsForm.APP_TITLE]: 'should be fetched from server',
  [valuesSettingsForm.LOGO_PATH]: 'should be fetched from server',
  [valuesSettingsForm.SELECT_THEME]: 'themeDefault',
  [valuesSettingsForm.BROWSER_TITLE]: 'should be fetched from server',
};

export { defaultPersonalFormValues, defaultContactFormValues, defaultAppSettingsFormValues }
