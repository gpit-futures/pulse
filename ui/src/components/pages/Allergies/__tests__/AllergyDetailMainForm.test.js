import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import configureStore from 'redux-mock-store';

import AllergyDetailMainForm from '../AllergiesDetail/AllergyDetailMainForm';
import { valuesNames, valuesLabels } from '../forms.config';

Enzyme.configure({ adapter: new Adapter() });

const mockStore = configureStore();
const store = mockStore({});
const FORM_NAME = 'allergiePanelForm';

const testProps = {
  detail: {
    dateCreated: 1507020019000,
  },
  isSubmit: false,
};

describe('Component <AllergyDetailMainForm />', () => {
  it('should renders with props correctly', () => {
    const component = shallow(
      <AllergyDetailMainForm
        store={store}
        isSubmit={testProps.isSubmit}
        detail={testProps.detail}
      />).dive().dive().dive();
    expect(component.find('Field')).toHaveLength(4);
    expect(component.find('form')).toHaveLength(1);
    expect(component.find('form').prop('name')).toEqual(FORM_NAME);

    expect(component.find('Field').at(0).props().name).toEqual(valuesNames.CAUSE);
    expect(component.find('Field').at(0).props().label).toEqual(valuesLabels.CAUSE);
    expect(component.find('Field').at(0).props().props.isSubmit).toEqual(false);

    expect(component.find('Field').at(1).props().name).toEqual(valuesNames.REACTION);
    expect(component.find('Field').at(1).props().label).toEqual(valuesLabels.REACTION);
    expect(component.find('Field').at(1).props().props.isSubmit).toEqual(false);

    expect(component.find('Field').at(2).props().name).toEqual(valuesNames.AUTHOR);
    expect(component.find('Field').at(2).props().label).toEqual(valuesLabels.AUTHOR);
    expect(component.find('Field').at(2).props().props.disabled).toEqual(true);

    expect(component.find('Field').at(3).props().name).toEqual(valuesNames.DATE);
    expect(component.find('Field').at(3).props().label).toEqual(valuesLabels.DATE);
    expect(component.find('Field').at(3).props().props.disabled).toEqual(true);
    expect(component.find('Field').at(3).props().props.value).toEqual(testProps.detail.dateCreated);
    expect(component.find('Field').at(3).props().props.format).toEqual('DD-MMM-YYYY');

    expect(component).toMatchSnapshot();
  });

  it('should renders with props correctly with is import', () => {
    const component = shallow(
      <AllergyDetailMainForm
        store={store}
        isSubmit
        detail={{
          dateCreated: 1507020019000,
          isImport: true,
        }}
      />).dive().dive().dive();
    expect(component.find('Field')).toHaveLength(5);

    expect(component.find('Field').at(2).props().name).toEqual(valuesNames.IMPORT);
    expect(component.find('Field').at(2).props().id).toEqual(valuesNames.IMPORT);
    expect(component.find('Field').at(2).props().label).toEqual(valuesLabels.IMPORT);
    expect(component.find('Field').at(2).props().props.isSubmit).toEqual(true);
    expect(component.find('Field').at(2).props().props.disabled).toEqual(true);

    expect(component).toMatchSnapshot();
  });

  it('should renders correctly when form is submitted', () => {
    const component = shallow(
      <AllergyDetailMainForm
        store={store}
        isSubmit
        detail={testProps.detail}
      />).dive().dive().dive();
    expect(component.find('Field')).toHaveLength(4);
    expect(component.find('form')).toHaveLength(1);
    expect(component.find('form').prop('name')).toEqual(FORM_NAME);

    expect(component.find('Field').at(0).props().props.isSubmit).toEqual(true);
    expect(component.find('Field').at(1).props().props.isSubmit).toEqual(true);

    expect(component).toMatchSnapshot();
  });
});

