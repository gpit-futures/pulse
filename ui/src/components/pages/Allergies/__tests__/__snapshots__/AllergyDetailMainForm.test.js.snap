// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <AllergyDetailMainForm /> should renders correctly when form is submitted 1`] = `
<div
  className="panel-body-inner"
>
  <form
    className="form"
    name="allergiePanelForm"
  >
    <div
      className="form-group-wrapper"
    >
      <div
        className="row-expand"
      >
        <div
          className="col-expand-left"
        >
          <Field
            component={[Function]}
            id="cause"
            label="Cause"
            name="cause"
            placeholder=""
            props={
              Object {
                "isSubmit": true,
              }
            }
            type="text"
          />
          <Field
            component={[Function]}
            id="reaction"
            label="Reaction"
            name="reaction"
            props={
              Object {
                "isSubmit": true,
              }
            }
          />
        </div>
      </div>
      <div
        className="row-expand"
      >
        <div
          className="col-expand-left"
        >
          <Field
            component={[Function]}
            id="author"
            label="Author"
            name="author"
            props={
              Object {
                "disabled": true,
              }
            }
          />
        </div>
        <div
          className="col-expand-right"
        >
          <Field
            component={[Function]}
            id="currentDate"
            label="Date"
            name="currentDate"
            props={
              Object {
                "disabled": true,
                "format": "DD-MMM-YYYY",
                "value": 1507020019000,
              }
            }
          />
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`Component <AllergyDetailMainForm /> should renders with props correctly 1`] = `
<div
  className="panel-body-inner"
>
  <form
    className="form"
    name="allergiePanelForm"
  >
    <div
      className="form-group-wrapper"
    >
      <div
        className="row-expand"
      >
        <div
          className="col-expand-left"
        >
          <Field
            component={[Function]}
            id="cause"
            label="Cause"
            name="cause"
            placeholder=""
            props={
              Object {
                "isSubmit": false,
              }
            }
            type="text"
          />
          <Field
            component={[Function]}
            id="reaction"
            label="Reaction"
            name="reaction"
            props={
              Object {
                "isSubmit": false,
              }
            }
          />
        </div>
      </div>
      <div
        className="row-expand"
      >
        <div
          className="col-expand-left"
        >
          <Field
            component={[Function]}
            id="author"
            label="Author"
            name="author"
            props={
              Object {
                "disabled": true,
              }
            }
          />
        </div>
        <div
          className="col-expand-right"
        >
          <Field
            component={[Function]}
            id="currentDate"
            label="Date"
            name="currentDate"
            props={
              Object {
                "disabled": true,
                "format": "DD-MMM-YYYY",
                "value": 1507020019000,
              }
            }
          />
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`Component <AllergyDetailMainForm /> should renders with props correctly with is import 1`] = `
<div
  className="panel-body-inner"
>
  <form
    className="form"
    name="allergiePanelForm"
  >
    <div
      className="form-group-wrapper"
    >
      <div
        className="row-expand"
      >
        <div
          className="col-expand-left"
        >
          <Field
            component={[Function]}
            id="cause"
            label="Cause"
            name="cause"
            placeholder=""
            props={
              Object {
                "isSubmit": true,
              }
            }
            type="text"
          />
          <Field
            component={[Function]}
            id="reaction"
            label="Reaction"
            name="reaction"
            props={
              Object {
                "isSubmit": true,
              }
            }
          />
        </div>
      </div>
      <Field
        component={[Function]}
        id="originalSource"
        label="Import URL"
        name="originalSource"
        props={
          Object {
            "disabled": true,
            "isSubmit": true,
          }
        }
      />
      <div
        className="row-expand"
      >
        <div
          className="col-expand-left"
        >
          <Field
            component={[Function]}
            id="author"
            label="Author"
            name="author"
            props={
              Object {
                "disabled": true,
              }
            }
          />
        </div>
        <div
          className="col-expand-right"
        >
          <Field
            component={[Function]}
            id="currentDate"
            label="Date"
            name="currentDate"
            props={
              Object {
                "disabled": true,
                "format": "DD-MMM-YYYY",
                "value": 1507020019000,
              }
            }
          />
        </div>
      </div>
    </div>
  </form>
</div>
`;
